AMQPC-14: Added mTLS support (#1200)
mulesoft/docs-connectors: amqp/1.7/modules/ROOT/pages/amqp-mtls.adoc

> = Handle Message Acknowledgment - Mule 4

Reviewer: is this correct?

:page-aliases: connectors::amqp/amqp-mtls.adoc

Anypoint Connector for AMQP provides mTLS connection capabilities.

== Connection setup

In order to connect to an AMQP broker with mutual TLS (mTLS) authenthication enabled you need the following items:
* Client Truststore including the Certificate Authority's certificates required for server validation.
* Client Keystore providing the client private key and certificate required to authentication.

The options useTls and useSasl must be enabled.

`[source,xml,linenums]
----
<amqp:connection host="localhost" port="${rabbitmq.tls.port}" useTls="true" useSasl="true" >
            <tls:context >
                <tls:trust-store path="mtls/truststore.jks" password="test" type="jks" />
                <tls:key-store type="jks" path="mtls/test.jks" alias="test" keyPassword="test" password="test" />
            </tls:context>
</amqp:connection>
----
`

> == See Also

Reviewer: Wouldl be good to link to this doc https://docs.mulesoft.com/mule-runtime/4.3/tls-configuration

= Configure the Mutual TLS Authentication - Mule 4
:page-aliases: connectors::amqp/amqp-mtls.adoc

The following example shows how to configure Anypoint Connector for AMQP (AMQP Connector) to connect to an AMQP broker with mutual TLS (mTLS) authentication. To configure the global configuration connection you need:

> * A client Trust Store that includes the Certificate Authority's certificates required for the server validation.

Reviewer 2: ```suggestion
* A client truststore that includes the Certificate Authority's certificates that are required for the server validation.
```

> * A client Key Store that provides the client private key and certificate required for the authentication.

Reviewer 2: ```suggestion
* A client keystore that provides the client private key and certificate that are required for authentication.
```

> * The parameters Use TLS and Use SNI must be enabled on the AQMP global configuration in Anypoint Studio.

Reviewer 2: ```suggestion
* To enable the parameters *Use TLS* and *Use SNI* on the AQMP global configuration in Anypoint Studio.
```

== Configure the Connection

To configure the Mutual TLS authentication connection in Studio, follow these steps:

. Navigate to the *Global Elements* tab and click *Create*.
. In the filter box, type `AMQP`.
. Select *AQMP Config*.
. Click *OK*.
. In the *General* tab, for the *Connection* section, set the following parameters:
+
* *Host*: `localhost`
* *Port*: `${rabbitmq.tls.port}`
+

> . Select the *Use tls* and *Use sni* parameters box.

Reviewer 2: ```suggestion
. Select the *Use tls* and *Use sni* parameters boxes.
```

. Click the *TLS* tab.

> . For *TLS Configuration* select `Edit inline`.

Reviewer 2: ```suggestion
. For *TLS Configuration*, select *Edit inline*.
```
